mobile/lib/interfaces/network.interface.dart

```diff
@@ -0,0 +1,8 @@
+abstract interface class INetworkRepository {
+  Future<String?> getWifiName();
+  Future<String?> getWifiIp();
+
+  Future<bool> isWifiConnected();
+  Future<bool> isVpnConnected();
+  Future<bool> isMobileDataConnected();
```

Comment:
Only `isWifiConnected` is used. Do we actually need the others? Also, what happens if you just call `getWifiName` without checking `isWifiConnected`? It'd be better to do that so maybe `isWifiConnected` isn't needed either unless you intend to use it elsewhere.